pageserver/src/thread_mgr.rs

```diff
@@ -0,0 +1,297 @@
+//!
+//! This module provides centralized handling of threads in the Page Server.
+//!
+//! We provide a few basic facilities:
+//! - A global registry of threads that lists what kind of threads they are, and
+//!   which tenant or timeline they are working on
+//!
+//! - The ability request a thread to shut down.
+//!
+//! TODO: This would be a good place to also handle panics in a somewhat sane way
+//! Depending on what thread panics, we might want to kill the whole server, or
+//! only a single tenant or timeline.
+//!
+
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::panic;
+use std::panic::AssertUnwindSafe;
+use std::sync::atomic::{AtomicBool, AtomicU64, Ordering};
+use std::sync::{Arc, Mutex};
```

Comment:
> `Mutex`

Do you think we can reuse parking lot here too?

Reply:
Sure, although std::sync::Mutex is still what most places use. I'm in favor of switching the whole codebase to parking_lot, but I'd prefer to do it as one separate commit.